providers/snowflake/warehouse_grant.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package snowflake
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+)
+
+type WarehouseGrantGenerator struct {
+	SnowflakeService
+}
+
+func (g WarehouseGrantGenerator) createResources(warehouseGrantList []warehouseGrant) []terraform_utils.Resource {
+	type tfGrant struct {
+		DB        string
+		Privilege string
+		Roles     []string
+		Shares    []string
+	}
+	groupedResources := map[string]*tfGrant{}
+	for _, grant := range warehouseGrantList {
+		// TODO(ad): Fix this csv delimited when fixed in the provider. We should use the same functionality.
+		id := fmt.Sprintf("%s|||%s", grant.Name.String, grant.Privilege.String)
+		_, ok := groupedResources[id]
+		if !ok {
+			groupedResources[id] = &tfGrant{
+				DB:        grant.Name.String,
+				Privilege: grant.Privilege.String,
+				Roles:     []string{},
+				Shares:    []string{},
+			}
+		}
+		tfGrant := groupedResources[id]
+		if grant.GrantedTo.String == "ROLE" {
```

Comment:
I also wonder if we should change these to be a switch. Might be worth updating these everywhere
```
switch grant.GrantedTo.STring
case "ROLE":
case "SHARE":
default:
   this should error out so we are aware of different types of grants
```